### PR TITLE
feat(email): add a pre-header option to email template

### PR DESCRIPTION
**Pull request by claubatista**
## Because

- Registration email pre-header is not very informative.

## This pull request

- Add a pre-header option to email template.
- Use the pre-header in the verify short code email. 

## Issue that this pull request solves

Closes: #7973

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Captura de Tela 2021-04-05 às 17 45 51

## Other information (Optional)

Hey folks, I have some questions about the changes. 

1. I didn't change the `.txt` file, because I didn't know if it was necessary. 
2. Should I add pre-headers in other emails? 
3. I was reading in this [blog post](https://www.activecampaign.com/blog/email-preheader) about pre-headers and it says it is good if the pre-header has 40-130 characters long. 
Should I add validation around this? 

Cheers!

### CODE REVIEW

File: packages/fxa-auth-server/lib/senders/templates/layouts/fxa.html
Comment (claubatista) [inline packages/fxa-auth-server/lib/senders/templates/layouts/fxa.html:18]: I know this is a little strange, but I based myself in the Netflix and Stripe promotional emails that I had in my inbox.
This is necessary to show only the pre-header and not the pre-header + the body email content. 

I did some tests around how much empty space is necessary in different screen widths and this is what I arrived at.

Comment (jodyheavener) [reply packages/fxa-auth-server/lib/senders/templates/layouts/fxa.html:18]: Emails are so fun.

**Review by jodyheavener — COMMENTED** [review]
> I didn't change the .txt file, because I didn't know if it was necessary.

I would say probably not, since the preheader in a text-based email would not be visibly hidden.

> Should I add pre-headers in other emails?

For now let's just do this email. Product can decide if we want to expand this. #8148

> I was reading in this blog post about pre-headers and it says it is good if the pre-header has 40-130 characters long. Should I add validation around this?

This is interesting. I wouldn't say it's not strictly necessary, as the copy will likely have been written by the time it gets to engineering, so it would need to be communicated further up. If you wanted a fun little exercise in validating emails by all means go for it, but I wouldn't block merging for it. It would almost be better suited to a linter or test warning than a straight up test failure. I'll let you decide!

**Comment by claubatista** [comment]
> This is interesting. I wouldn't say it's not strictly necessary, as the copy will likely have been written by the time it gets to engineering, so it would need to be communicated further up. If you wanted a fun little exercise in validating emails by all means go for it, but I wouldn't block merging for it. It would almost be better suited to a linter or test warning than a straight up test failure. I'll let you decide!

I was looking how to [implement a rule](https://eslint.org/docs/developer-guide/working-with-rules) in eslint and I think it would be overkill in this case.

If I were to implements a warning in the tests, where exactly do you suggest I do that? I looked in the tests and I couldn't find any tests for the email templates. So, I was a little confused.

**Comment by jodyheavener** [comment]
@claubatista check out `packages/fxa-auth-server/test/local/senders/email.js` for email tests. You can probably add one [here](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/test/local/senders/email.js#L1224:L1251), but you'll have to dig in to see if we already have a way to assert content length. This might be tricky with how our tests are set up. Refer to [these docs](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/README.md#testing) for how to run auth-server tests.

**Comment by claubatista** [comment]
Hi @jodyheavener, I looked at the tests and I think the less disruptive way to do it is using a regex. 
The regex would check how many characters there are inside the pre-header div before the spaces (assuming we include a `pre-header` class to it). This regex will only match if there is between 1 - 130 characters in this space.

```js
{ test: 'notMatch', expected: /(?:class="pre-header">\s*).{1,130}(?:\s*&nbsp;)/i },
```

The problem with this approach is we will not have a warning, but a failure. So, we could add a third propriety in the tests cases `warning: true`, and when this is true we warn instead of assert. 

IMHO, I don't think this is worth it. I noticed in my cellphone the pre-header doesn't show completely, even if it is short. 
Any number of character we put as a limit would be ad hoc at this moment, I feel like. WDYT?

**Review by jodyheavener — APPROVED** [review]
@claubatista Thanks for investigating this so thoroughly! I would agree with your conclusion. This LGTM. 🥳

**Pull request merged**
